backend/apps/monkvision/apis/lib/nlp.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * NLP Model and Predictions
+ *  
+ * (C) 2022 TekMonks. All rights reserved.
+ * License: See enclosed license.txt file.
+ */
+
+const fs = require("fs");
+const Mustache = require("mustache");
+const utils = require(`${APP_CONSTANTS.LIB_DIR}/nlp/utils.js`);
+const tf = require("@tensorflow/tfjs");
+const use = require("@tensorflow-models/universal-sentence-encoder");
+require('@tensorflow/tfjs-node');
+
+// Load the englishModel
+const englishModel = require(`${APP_CONSTANTS.CONF_DIR}/englishModel.json`);
+
+// Global declarations
+const MODELS = {}, TENSORS = {}, INTENTS = {}, METRICS = Object.assign({}, englishModel.metrics), 
+    DURATIONS = Object.assign({}, englishModel.durations), RESOURCES = Object.assign({}, englishModel.resources);
+
+// Train Model (JSON)
+INTENTS["metrics"] = JSON.parse(Mustache.render(fs.readFileSync(`${APP_CONSTANTS.CONF_DIR}/models/metric_intents.json`, "utf8"), englishModel));
+INTENTS["durations"] = JSON.parse(Mustache.render(fs.readFileSync(`${APP_CONSTANTS.CONF_DIR}/models/duration_intents.json`, "utf8"), englishModel));
+INTENTS["resources"] = JSON.parse(Mustache.render(fs.readFileSync(`${APP_CONSTANTS.CONF_DIR}/models/resource_intents.json`, "utf8"), englishModel));
+
+exports.init = async () => {
+    MODELS["encoder"] = use.load();
+    _setInputTensor();      // Set input tensors for all the intents
+
+    _createModelAndItsLayer("metrics", 3, englishModel.metrics.length, 200, _trainModel);
+    _createModelAndItsLayer("durations", 3, englishModel.durations.length, 250, _trainModel);
+    _createModelAndItsLayer("resources", 3, englishModel.resources.length, 200, _trainModel);
+}
+
+/** 
+ * Create the intents model and train the dataset
+ * @param {string} modelName - Intent Model name
+ * @param {number} layersCount - Layers to be added to the current Model
+ * @param {number} inputShape - Assign Input Shape to the Model
+ * @param {number} iterationCount - Model dataset training based on iteration count
+ * @param {function} callback - After Create and Train the model with Intents(json)
+ */
+function _createModelAndItsLayer(modelName, layersCount, inputShape, iterationCount, callback) {
+    MODELS[modelName] = tf.sequential();
+    for (let layer = 1; layer <= layersCount; layer++) {
+        MODELS[modelName].add(tf.layers.dense({
+            inputShape: [layer == 1 ? 512 : inputShape],
+            activation: 'sigmoid',
+            units: inputShape,
+        }));
+    }
+
+    MODELS[modelName].compile({
+        loss: 'meanSquaredError',
+        optimizer: tf.train.adam(.06), // This is a standard compile config
+    });
+    if (callback) callback(modelName, iterationCount);
```

Comment:
Why would we ever have callback in a pure CPU bound function with no other callbacks or awaits???